src/decimal.rs

```diff
@@ -1514,6 +1514,122 @@ impl Decimal {
             flags: flags(self.is_sign_negative(), quotient_scale),
         })
     }
+
+    /// The estimated exponential function, e<sup>x</sup>, rounded to 8 decimal places. Stops
+    /// calculating when it is within tolerance is roughly 0.000002 in order to prevent
+    /// multiplication overflow.
+    fn exp(&self) -> Decimal {
+        if self == &Decimal::zero() {
+            return Decimal::one();
+        }
+
+        let tolerance = Decimal::new(2, 7);
+        let mut term = self.clone();
+        let mut result = self.clone() + Decimal::one();
+        let mut prev_result = Decimal::zero();
+        let mut factorial = Decimal::one();
+        let mut n = Decimal::new(2, 0);
+
+        // Needs rounding because multiplication overflows otherwise.
+        while (result - prev_result).abs() > tolerance && n < Decimal::new(24, 0) {
```

Comment:
This could be optimized quite a bit by preventing the boxing/unboxing. We should probably make the new function a `const fn` to hint to the compiler too.

Reply:
I'm actually not quite sure how to properly optimize these to prevent the boxing problems. Do you have any resources on how that works? I'd love to learn in more detail and my googling hasn't been finding anything.